plutus-core/plutus-core/src/Data/SatInt.hs

```diff
@@ -49,49 +49,83 @@ instance Read SatInt where
   readsPrec p xs = [ (SI x, r) | (x, r) <- readsPrec p xs ]
 
 instance Eq SatInt where
-  SI x == SI y = eqInt x y
-  SI x /= SI y = neInt x y
+  {-# INLINE (==) #-}
+  (==) = coerce eqInt
+
+  {-# INLINE (/=) #-}
+  (/=) = coerce neInt
 
 instance Ord SatInt where
-  SI x <  SI y = ltInt x y
-  SI x <= SI y = leInt x y
-  SI x >  SI y = gtInt x y
-  SI x >= SI y = geInt x y
+  {-# INLINE (<) #-}
+  (<) = coerce ltInt
+
+  {-# INLINE (<=) #-}
+  (<=) = coerce leInt
+
+  {-# INLINE (>) #-}
+  (>) = coerce gtInt
+
+  {-# INLINE (>=) #-}
+  (>=) = coerce geInt
 
 -- | In the `Num' instance, we plug in our own addition, multiplication
 -- and subtraction function that perform overflow-checking.
 instance Num SatInt where
-  (+)               = plusSI
-  (*)               = timesSI
-  (-)               = minusSI
+  {-# INLINE (+) #-}
+  (+) = plusSI
+
+  {-# INLINE (*) #-}
+  (*) = timesSI
+
+  {-# INLINE (-) #-}
+  (-) = minusSI
+
+  {-# INLINE negate #-}
   negate (SI y)
     | y == minBound = maxBound
     | otherwise     = SI (negate y)
+
+  {-# INLINE abs #-}
   abs x
-    | x >= 0        = x
-    | otherwise     = negate x
-  signum (SI x)     = SI (signum x)
+    | x >= 0    = x
+    | otherwise = negate x
```

Comment:
Why is it a custom definition? How does it handle near-maximum values? Needs a few tests at the very least.

Reply:
Basically this stuff all got copied from `safeint`... which sadly means I don't have good justifications for a bunch of things :/

Reply:
I remember thinking at the time that we should dump everything from `safeInt` except for the few things we actually need.

Reply:
I've checked it and I think the current behavior is as correct as the default one:

```
>>> (minBound :: SatInt, negate (minBound :: SatInt))
(-9223372036854775808,9223372036854775807)
>>> (minBound :: SatInt, 0 - minBound :: SatInt)
(-9223372036854775808,9223372036854775807)
 
>>> (maxBound :: SatInt, negate (maxBound :: SatInt))
(9223372036854775807,-9223372036854775807)
>>> (maxBound :: SatInt, 0 - maxBound :: SatInt)
(9223372036854775807,-9223372036854775807)
```

Maybe it's a custom definition to make it faster or clearer.

Would still be great to have such tests if they're not there.